### ISSUE TITLE
dateFile assumes config is mutable

### ISSUE THREAD

**Issue by OrangeDog** [log4js-node/log4js-node#743]
If it's been loaded from node-config, then it won't be, and throws:
```
TypeError: Cannot assign to read only property 'alwaysIncludePattern' of object '#<Object>'
    at Object.configure (./node_modules/log4js/lib/appenders/dateFile.js:50:33)
    at Configuration.createAppender (./node_modules/log4js/lib/configuration.js:95:39)
    at appenderNames.forEach (./node_modules/log4js/lib/configuration.js:130:47)
    at Array.forEach (<anonymous>)
    at Configuration.set appenders [as appenders] (./node_modules/log4js/lib/configuration.js:123:19)
    at new Configuration (./node_modules/log4js/lib/configuration.js:232:20)
    at Object.configure (./node_modules/log4js/lib/log4js.js:205:12)
```

**Comment by nomiddlename** [comment]
This is similar to previous issues relating to node-config. If someone wants to tackle this, start with a test for dateFile that freezes the config object.

**Comment by victor0801x** [comment]
I seems have fixed in 3.x ? @OrangeDog 

```js
// config/default.json
/*
{
	"dateFile": {
		"type": "dateFile",
		"filename": "./log",
		"pattern": "yyyyMMdd",
		"compress": false
	}
}
*/
"use strict";
const config = require("config"); // 1.30.0
const log4js = require("log4js"); // 3.0.2
log4js.configure({
	appenders: { dateFile: config.get("dateFile") },
	categories: {
	  default: { appenders: [ 'dateFile' ], level: 'debug' }
	}
});
const log = log4js.getLogger();
log.error("hi")
```

**Comment by OrangeDog** [comment]
@victor0801x the problem occurs when the entire config object is frozen. That's where it tries to mutate `alwaysIncludePattern`.
For example:
```js
log4js.configure(config.get('log4js'))
```

**Comment by nomiddlename** [comment]
Published to npm in version 3.0.3
